plugins/hypervisors/vmware/src/com/cloud/storage/resource/VmwareStorageProcessor.java

```diff
@@ -2265,7 +2265,11 @@ public Answer forgetObject(ForgetObjectCmd cmd) {
     }
 
     private static String deriveTemplateUuidOnHost(VmwareHypervisorHost hyperHost, String storeIdentifier, String templateName) {
-        String templateUuid = UUID.nameUUIDFromBytes((templateName + "@" + storeIdentifier + "-" + hyperHost.getMor().getValue()).getBytes("UTF-8")).toString();
+        try{
+            String templateUuid = UUID.nameUUIDFromBytes((templateName + "@" + storeIdentifier + "-" + hyperHost.getMor().getValue()).getBytes("UTF-8")).toString();
+        }catch(UnsupportedEncodingException e){
+            s_logger.warn("Unable to generate UUID due to unexpected encoding error."+e);
```

Comment:
@DaanHoogland 
I changed to localized message as requested.
I don't expect this exception to ever be thrown, because it will only do so if the undelying platform really doesn't support UTF-8 at all. Can't think of a platform that wouldn't support that nowadays, wether it's the default or not.
Adding a retry within the catch would require another catch within it, so i think its too much for an exception that would likely never be thrown ;)